# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/NotificationController.h
# Commit: 9f1d0d70629c
# Full Hash: 9f1d0d70629c54acefe940e5f93310126b681fd9
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2022-06-14 03:47:07
# Regressor Bug: 1758689
# File Overlap Count: 1
# Description:
#   Bug 1758689: Create and maintain a viewport cache for hittesting r=Jamie,emilio
#   
#   This hittesting approach is borrowed from our Android implementation.
#   We maintain a cache of ID's for accessibles which are visible in the viewport.
#   This cache is created with a call to `nsLayoutUtils::GetFramesForArea`, which
# ==============================================================================

diff -r b4f6fb035220 -r 9f1d0d70629c accessible/base/NotificationController.h
--- a/accessible/base/NotificationController.h	Mon Jun 13 22:11:11 2022 +0000
+++ b/accessible/base/NotificationController.h	Mon Jun 13 22:28:36 2022 +0000
@@ -91,7 +91,8 @@
  * Used to process notifications from core for the document accessible.
  */
 class NotificationController final : public EventQueue,
-                                     public nsARefreshObserver {
+                                     public nsARefreshObserver,
+                                     public nsAPostRefreshObserver {
  public:
   NotificationController(DocAccessible* aDocument, PresShell* aPresShell);
 
@@ -299,6 +300,7 @@
 
   // nsARefreshObserver
   virtual void WillRefresh(mozilla::TimeStamp aTime) override;
+  virtual void DidRefresh() override;
 
   /**
    * Set and returns a hide event, paired with a show event, for the move.